wasp-ai/src/client/pages/MainPage.jsx

```diff
@@ -95,19 +108,23 @@ const MainPage = () => {
 
   async function startGenerating(event) {
     event.preventDefault();
-
     try {
-      const appNum = JSON.parse(localStorage.getItem("appNum"))
-      localStorage.setItem("appNum", appNum + 1)
-      localStorage.setItem("appDetails", JSON.stringify({
-        appName,
-        appDesc,
-        appPrimaryColor: appPrimaryColor.name,
-        appAuthMethod: appAuthMethod.value,
-        appCreativityLevel: creativityLevel.value,
-      }));
+      localStorage.setItem(
+        "appDetails",
+        JSON.stringify({
+          appName,
+          appDesc,
+          appPrimaryColor: appPrimaryColor.name,
+          appAuthMethod: appAuthMethod.value,
+          appCreativityLevel: creativityLevel.value,
+        })
+      );
+      if (!user) {
+        setIsLoginModalOpen(true);
+        return;
+      }
     } catch (error) {
-      console.error(error)
+      console.error(error);
     }
```

Comment:
miho said localStorage methods often fails (quota exceeded, browser security restriction) so it's good practice to wrap it in a try-catch block to prevent it from crashing the app.

Reply:
Oh wow had no idea about that! Right, if that is so then it makes sense.